src/pilot/assign_role.py

```diff
@@ -399,8 +420,14 @@ def define_storage_logical_disks(drac_client, raid_controller_name):
         return None
 
     # Define a logical disk to host the operating system.
-    os_logical_disk = define_storage_operating_system_logical_disk(
-        raid_cntlr_physical_disks, raid_controller_name)
+    raid_ctrl_name = raid_controller_name
+    raid_disks = raid_cntlr_physical_disks
+    boss_dsk_info = is_boss_controller(raid_disks, raid_ctrl_name, drac_client)
+    if boss_dsk_info:
+        os_logical_disk = boss_dsk_info
+    else:
+        os_logical_disk = define_storage_operating_system_logical_disk(
```

Comment:
Let's remove the above added lines and just leave this as a call to define_storage_operating_system_logical_disk()

Reply:
okay

Reply:
To skip the call, define_storage_operating_system_logical_disk() we added above code. 
Reason to skip that is - below lines in the method - 
os_logical_disk_size_gb,
     os_physical_disk_names) = find_physical_disks_for_storage_os(
        physical_disks)